### DISCUSSION TITLE
Experience of RStudio User Trying to migrate

### DISCUSSION THREAD

**Discussion by david-priest** [REditorSupport/vscode-R#1750] [General]
I am a data scientist.  I was recommended to try Visual Studio Code as a replacement for RStudio.  I gave it a good shot but decided not to use it for the following reasons:

1) I am used to RStudio Code Colouring and was not able to reproduce it in Visual Studio.
2) Outline works only half the time and I couldn't find an option to just show a simplified version like in RStudio (only labels lines starting with # outside chunks.)
3) Plots/Outputs do not show up inline in the R notebook like in Rstudio.  There's no way to quickly copy a plot by right clicking.
4) Variables did not show up.  I could not double click variables (like a dataframe) to check its contents.

Overall, I think you should address these basic issues to make it easier to migrate from RStudio.  It shouldn't be a matter of searching through this issue forum for 2hours+ to get these basic things to work.

**Comment by albertosantini** [comment]
Apart the first point, there are a lot of themes outside to choose, the rest of points here work.

2. You should give a look at `R` panel in the Activity Bar. It contains a Workspace panel with Attached Namespaces, Loaded Namespaces and Global Environment.
3. When I plot (or ggplot) something, a plot is displayed inside an editor: right clicking displays a menu with `copy` item.
4. See 2): hovering on the variable you may remove or view the object; if you view the object a panel is displayed with the content of the variable; if the variable is a data frame, you see a sort of spreadsheet.

Hope that helps; just in case you may provide more information or steps to reproduce the issue.

**Comment by benz0li** [comment]
> I was recommended to try Visual Studio Code as a replacement for RStudio.

The beauty of VS Code is that it has extensions for [R](https://marketplace.visualstudio.com/items?itemName=REditorSupport.r), [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python) and [Julia](https://marketplace.visualstudio.com/items?itemName=julialang.language-julia) – today's major Data Science languages.

@purisuto VS Code not only replaces RStudio but also [PyCharm](https://www.jetbrains.com/pycharm/) and [Juno](https://junolab.org); the latter has been officially replaced by the VS Code Julia extension.

---

Check out my *reference deployment* at https://demo.jupyter.b-data.ch or have a look at

* [JupyterLab R docker stack](https://github.com/b-data/jupyterlab-r-docker-stack)
* [JupyterLab Python docker stack](https://github.com/b-data/jupyterlab-python-docker-stack)
* [JupyterLab Julia docker stack](https://github.com/b-data/jupyterlab-julia-docker-stack)

ℹ️ Each docker stack has also GPU accelerated images available.

---

What makes these images different:

1. Multi-arch: `linux/amd64`, `linux/arm64/v8`
  ℹ️ Runs on Apple M series using Docker Desktop.
1. Base image: [Debian](https://hub.docker.com/_/debian) instead of [Ubuntu](https://hub.docker.com/_/ubuntu)
  ℹ️ CUDA-enabled images are Ubuntu-based.
1. IDE: [code-server](https://github.com/coder/code-server) next to [JupyterLab](https://github.com/jupyterlab/jupyterlab)
  ℹ️ code-server = VS Code in the browser.
1. Just Python – no [Conda](https://github.com/conda/conda) / [Mamba](https://github.com/mamba-org/mamba)

**Comment by LR1933** [comment]
> > I was recommended to try Visual Studio Code as a replacement for RStudio.
> 
> The beauty of VS Code is that it has extensions for [R](https://marketplace.visualstudio.com/items?itemName=REditorSupport.r), [Python](https://marketplace.visualstudio.com/items?itemName=ms-python.python) and [Julia](https://marketplace.visualstudio.com/items?itemName=julialang.language-julia) – today's major Data Science languages.
> 
> @purisuto VS Code not only replaces RStudio but also [PyCharm](https://www.jetbrains.com/pycharm/) and [Juno](https://junolab.org); the latter has been officially replaced by the VS Code Julia extension.
> 
> Check out my _reference deployment_ at https://demo.jupyter.b-data.ch or have a look at
> 
> * [JupyterLab R docker stack](https://github.com/b-data/jupyterlab-r-docker-stack)
> * [JupyterLab Python docker stack](https://github.com/b-data/jupyterlab-python-docker-stack)
> * [JupyterLab Julia docker stack](https://github.com/b-data/jupyterlab-julia-docker-stack)
> 
> ℹ️ Each docker stack has also GPU accelerated images available.
> 
> What makes these images different:
> 
> 1. Multi-arch: , `linux/amd64``linux/arm64/v8`
>    ℹ️ Run natively on Apple M1/M2 (Docker Desktop).
> 2. Base image: [Debian](https://hub.docker.com/_/debian) instead of [Ubuntu](https://hub.docker.com/_/ubuntu)
>    ℹ️ CUDA-enabled images are Ubuntu-based.
> 3. IDE: [code-server](https://github.com/coder/code-server) next to [JupyterLab](https://github.com/jupyterlab/jupyterlab)
>    ℹ️ code-server = VS Code in the browser.
> 4. Just Python – no [Conda](https://github.com/conda/conda) / [Mamba](https://github.com/mamba-org/mamba)

VS Code is great, but the biggest problem for me is that there is no color distinction for the input and output of R code in the terminal. Additionally, the error and warning messages are also written similarly to other code. So I have to face a large white undifferentiated code in terminal when using VS Code. Rstudio does great about this. Do you have any advise to solve this problems?

**Comment by eitsupi** [comment]
@lijiaqi-github Have you tried radian? https://github.com/randy3k/radian

**Comment by LR1933** [comment]
> @lijiaqi-github Have you tried radian? https://github.com/randy3k/radian

Sure, but there is a new problem, [https://github.com/REditorSupport/vscode-R/issues/1390](https://github.com/REditorSupport/vscode-R/issues/1390).

I'm sorry to say this, but I think Radian is a bit slow to respond, and there are some errors that I have never encountered in RStudio.

**Comment by benz0li** [comment]
> Sure, but there is a new problem, #1390.

@lijiaqi-github Can you reproduce this on my *reference deployment* at https://demo.jupyter.b-data.ch?

- OS: Debian 12
- Code Version: 1.79.2
- Radian Version: 0.6.6
  - r executable: `/usr/local/lib/R/bin/R`
  - r version: 4.3.1
  - python executable: `/usr/local/bin/python`
  - python version: 3.11.4

ℹ️ This seems to be a Windows specific issue.

**Comment by LR1933** [comment]
@benz0   
li I'm not very familiar with programming software, so I'm hoping that's my problem. It was me who made a mistake, but I can't figure out the reason. 

Would it be better to switch from Windows to Mac?

Here is my configuration and VScode setting.json. I tried to open Radian by VS Code in your Jupyter link, but I'm not familiar with Jupyter and couldn't succeed. I haven't used Jupyter before, so I don't know how to do it.
#########################################################################
OS: Windows 11 Pro for Workstations version-22H2 22621.1928
Code Version: 1.80.0 (VS code)
Radian Version: 0.6.6 (I don't know how to check the version. I installed radian last week so it should be 0.6.6)
r executable: C:\Program Files\R\R-4.3.1\bin\R.exe
r version: 4.3.1
python executable: C:\Users\[myname]\AppData\Local\Programs\Python\Python311\python.exe
python version: 3.11.4
VS code setting.json
{
    "r.bracketedPaste": true,
    "r.rterm.windows": "C:\\Users\\[myname]\\AppData\\Local\\Programs\\Python\\Python311\\Scripts\\radian.exe",
}
#########################################################################

[a comment by an automated account is not shown]

**Comment by ThomasSoeiro** [comment]
@eitsupi @Fred-Wu perhaps this "issue" can be closed (no actual issues reported) or moved to https://github.com/REditorSupport/vscode-R/discussions (idk if it is possible)
